### ISSUE TITLE
[service] [request]: describe request here

### ISSUE THREAD

**Issue by Byeong-jun-Kim** [aws/containers-roadmap#1242]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
I want to use logging to fluentd(forward output) or logstash(http output)

**Which service(s) is this request for?**
EKS on Fargate

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
Only CloudWatch, ES, Kinesis are available now as basic EKS Fargate fluent bit functions.

**Are you currently working around this issue?**
I added a fluent-bit container in every pods.

**Additional context**
We hope that the general http & forward output function will be implemented as soon as possible in order for EKS Fargate to become a more practical service.

**Attachments**

**Comment by mhausenblas** [comment]
Thanks for raising this issue @Byeong-jun-Kim. Could you please expand a bit on what you'd like to see beyond what we described in [Fluent Bit for Amazon EKS on AWS Fargate is here](https://aws.amazon.com/blogs/containers/fluent-bit-for-amazon-eks-on-aws-fargate-is-here/)?

**Comment by Byeong-jun-Kim** [comment]
Thank you for your reply.
The document says: 'You can choose between CloudWatch, ElasticSearch, Kinesis Firehose and Kinesis Streams as outputs.'
I feel that the output support above is insufficient.
Many Kubernetes users collect and process logs from filebeat to logstash or from fluent bit to fluentd, and send them to various places.
Besides that, it would be great to have Forward which sent to the same stack, fluentd, and http that can be used in general.
Is my request a little clearer? @mhausenblas

**Comment by pmbeaumier** [comment]
@mhausenblas Adding another example here.
In our case the HTTP output mentioned by @Byeong-jun-Kim would be great to have as we're using SumoLogic for our log system and this is the recommended way to send logs to SumoLogic. The [Fluentbit documentation](https://docs.fluentbit.io/manual/pipeline/outputs/http#example-sumo-logic-http-collector) actually provides an example.
Currently we have to go through a [cumbersome configuration](https://help.sumologic.com/03Send-Data/Collect-from-Other-Data-Sources/Amazon-CloudWatch-Logs#Create_a_stack_on_the_AWS_CloudFormation_console) with Cloudwatch in-between.
Thanks

**Comment by Dametrain** [comment]
To add to this I would like to see all native Fluentbit outputs as documented here: https://docs.fluentbit.io/manual/v/1.5/pipeline/outputs/. To start Datadog and Splunk would be my top two. As a work around you can use Kinesis Firehose, however that just adds cost and complexity.

**Comment by prashilgupta** [comment]
+1

**Comment by amitkatyal** [comment]
+1

**Comment by lennartt** [comment]
It would be ideal if we can send the logs to a HTTP output to Logstash.

**Comment by philomory** [comment]
Yeah, this limitation basically prevents us from using Fargate's logging functionality at all, until the GELF output is supported.

**Comment by philomory** [comment]
Yeah, this limitation basically prevents us from using Fargate's logging functionality at all, until the GELF output is supported.

I'm curious, @Byeong-jun-Kim, how did adding a fluent-bit container to every pod work out for you? It seems to me like you wouldn't be able to have that pod access the "node"'s filesystem to read the logs. Or are you just have your pods log to a file on a e.g. emptyDir volume so that it can be read by the fluent-bit pod?

**Comment by Byeong-jun-Kim** [comment]
@philomory yes you're right
We are processing the log with a fluent-bit by redirecting all the logs as a file. Since the storage space is limited, we designated a specific lifecycle by saving it by date.
But it's very tricky and inefficient. That's why I made this request.

**Comment by fishpen0** [comment]
Grafana shops would be ecstatic for loki support.

**Comment by segler-alex** [comment]
i need output to loki
+1

**Comment by sburns** [comment]
Would like to see the [DataDog output](https://docs.fluentbit.io/manual/pipeline/outputs/datadog) supported as well. Currently can work around by Cloudwatch -> DataDog forwarder -> DataDog, but that's money I'd just rather not spend.

**Comment by FelipeLujan** [comment]
Greetings.

At our company, we use Coralogix for log storage and analysis. 

Our logs get centralized in a Kafka cluster (confluent) and then consumed by an app that formats and forwards them to Coralogix.

The following config

```
    [OUTPUT]
        Name                                  kafka
        Match                                 *
        Brokers                               {{ .Values.logging.kafkaEndpoint }}:{{ .Values.logging.kafkaPort }}
        Topics                                test-topic
        Topic_Key                             topic
        Format                                json
        rdkafka.security.protocol             SASL_SSL
        rdkafka.sasl.mechanism                PLAIN
        rdkafka.sasl.username                 xxxx
        rdkafka.sasl.password                 xxxx
        rdkafka.queue.buffering.max.messages  500000
```

causes the following error
`"0500-amazon-eks-fargate-configmaps-admission.amazonaws.com" denied the request: kafka is not a supported output plugin. Please fix the logging configmap`


Please add support for Kafka outputs

**Comment by zquintana** [comment]
We too would really appreciate the Datadog and Http plugins, is that on the roadmap?

**Comment by pitstop-sirish-bajpai** [comment]
+1

loki plugin support on EKS fargate fluentbit

**Comment by tloriato** [comment]
+1, fundamental for us

**Comment by marcusdb** [comment]
+1

loki plugin support on EKS fargate

**Comment by kintoandar** [comment]
Could we please have Loki support for Fargate logs?

Please 🙏

**Comment by DevSusu** [comment]
@mhausenblas there's quite a reaction in this thread, would you be willing to take this issue further?
Any kind of response will be appreciated, it's been over an year after you commented.

**Comment by rielas** [comment]
S3 please 🙏

**Comment by eyalle** [comment]
Splunk please! 🙏

**Comment by teochenglim** [comment]
Oh dear, we are using ec2 node and considering switching to Fargate, however the current support of fluent-bit is limited and almost useless.

1. It doesn't support s3, so the log is not long term and cheap? Or we need to manually export from Cloudwatch to s3? That means we can only use Fargate as toy and not production ready?

2. For almost all of us have our existing log system, that means we need additional label or minimum mutation label on our log in a multiple environment settings.

3. Since everyone would likely need the Kubernetes metadata parser, wouldn't it be nice if default already parse? yet customer has minimum control?

4. What choice do we have now? Maybe i should plugin a fluentd side-car (500M binary) on my golang microservice (20M binary) just to support this limitation.

**Comment by ianrodrigues** [comment]
No more updates on this one? HTTP output would be great.

**Comment by singhnix** [comment]
Hello,

One of my customer have below requirements as well. Kindly add this to the feature request as well:


Currently , Fargate provides a hook for logging with fluentbit: https://docs.aws.amazon.com/eks/latest/userguide/fargate-logging.html . 

Problem: AWS only exposes very specific plugins. https://github.com/aws/containers-roadmap/issues/1242 . 

We need the fluent forward output plugin in order to get fluentbit fargate logs to Grafana Enterprise Logs (GEL).Fargate validates against the following supported output: es, firehose, kinesis_firehose, cloudwatch, cloudwatch_logs, and kinesis. 

Request to AWS is to  add “Fluent forward output plugin” to the above list

**Comment by junowong0114** [comment]
Kudos to @Byeong-jun-Kim. Adding the "http" or "forward" output can tremendously improve the logging experience on EKS Fargate.

My organization has an existing ECS/lambda -> Fluentd on ECS -> on-premise Elasticsearch architecture for some reasons.

1. Fluentd can congregate logs into batch requests before sending them to Elasticsearch, which greatly reduces network/CPU loading.
2. We can enforce centralized data enrichment or manipulation rules. We can also do that by Elasticsearch ingest pipeline, but then we would lose the other advantages.
3. Fluentd provides an extra layer of buffer between microservices and Elasticsearch so that in the unfortunate event of a network/Elasticsearch outage, the application logs are properly handled by Fluentd instead of drowning the microservices' resources.
4. There is of course the cost-saving factor as we managed our on-premise Elasticsearch cluster.

All of these are not possible because the "forward" and "http" output filters are disabled. Now I change the code to log to both stdout and to a log file, mount that file to Fluent Bit sidecar, and add another sidecar to perform log rotation... It is a nightmare and is seriously hindering the adoption of AWS EKS on Fargate within my organization.

Please fix this, or at least let us know if "forward" and "http" output filter will be considered to be added.

**Comment by rafilkmp3** [comment]
Guys do you think will be possible to configure my promtail to scrap logs from fargate namespace?

**Comment by jordanbcooper** [comment]
Still looking for Loki ouput support!

**Comment by sameeraksc** [comment]
datadog please

**Comment by VeresMarcell** [comment]
+1

Loki support would be great

**Comment by teimor** [comment]
any HTTP support would be great

**Comment by Rudya93** [comment]
+1

Loki support would be great

**Comment by gespi1** [comment]
What is the status of this? There has not been any word from the official maintainers in awhile, almost abandoned.

Looking for Loki plugin support and any kind of update. Ty

**Comment by Rudya93** [comment]
> 

+1

**Comment by trylix** [comment]
+1

Loki support would be great

**Comment by codesenju** [comment]
+ 1

**Comment by prein** [comment]
> At our company, we use Coralogix for log storage and analysis.

Did you try using elasticsearch output to ship directly to Coralogix? Did it work? If yes, would you mind sharing how to? Thanks!

**Comment by sonykphilip** [comment]
+1 for Loki

**Comment by smitb** [comment]
+1 for Loki

**Comment by asychev** [comment]
+1 for Loki

**Comment by chas0amx** [comment]
I attempted to forward logs from Karpenter, running on an AWS Fargate node, to [fargate-eks-sidecar-injector](https://github.com/mziyabo/fargate-eks-sidecar-injector) or any other destination and encountered the following error:

`panic: open sink "https://aos-xxx.us-east-1.es.amazonaws.com/": no sink found for scheme "https"
goroutine 1 [running]:
github.com/samber/lo.must({0x25adc00, 0xc00090c160}, {0x0, 0x0, 0x0})`

I observed that the logger seems to have functionality related [to parsing URLs](https://github.com/uber-go/zap/blob/d27427d23f81dba1f048d6034d5f286572049e1e/sink.go#L130).

Does anyone know the correct solution in this case or how to correctly forward logs to an external endpoint (rather than to the filesystem)?

**Comment by billnbell2** [comment]
Sematext would be good. 

Otis Gospodnetic <otis@sematext.com>

**Comment by hjweddie** [comment]
+1 for Loki

**Comment by ojohn** [comment]
Another one for Loki

**Comment by jeffreydlcheung** [comment]
+1 for supporting "Forward" and also other output listed fluentbit manual

**Comment by sharraj** [comment]
Many users asked for this "HTTP support would be great". Appreciate if this can be prioritized as it will open up ways to connect to way more end points.

**Comment by Mkori1** [comment]
Forward output plugin would be great.

**Comment by Th3D0ct0r** [comment]
Is this beeing considered? I cannot find anything regarding loki, forward or http on the roadmap. My and my customers problem would be solved by supporting loki as an output.

**Comment by RichardoC** [comment]
The opentelemetry plugin, or HTTP plugin being supported would be great for sending to dash0

**Comment by scottux** [comment]
Loki. Otel. +1

Please. I don't want to make a sidecar, or go to Kinesis and then back out to an agent and then to Loki.

**Comment by moalam93** [comment]
+1 where is this feature? Its been years

**Comment by andrewloux** [comment]
me in the quarterly review trying to explain why our logging costs as much as our entire compute spend: "listen. was i in a good place mentally when i suggested eks on fargate? no. was i gaslit by aws marketing? yes. do i regret everything? absolutely. hotel? trivago."

**Comment by lacherogwu** [comment]
+1 for Loki

**Comment by RichardoC** [comment]
> The opentelemetry plugin, or HTTP plugin being supported would be great for sending to dash0

For anyone wondering, I worked around this by using the ES plugin to write to them, but http/otel would be so much better

**Comment by lacherogwu** [comment]
@RichardoC, can you share more details about how you managed to work around this?

**Comment by RichardoC** [comment]
Here's an example output.conf I'm using which uses the fact dash0 can accept data sent as if it was elastic search. 

[OUTPUT]
    Name        es
    Match       *
    Host        some_ingress
    Port        443
    TLS         on
    HTTP_User   Bearer
    HTTP_Passwd some_token
    Index       some_dataset

Here are the (customer only) docs about this from dash0 <https://app.dash0.com/hub/integration/int_aws_eks_on_fargate/overview>

Here's the one line reference I can find, to the fact elastic is supported as an output (opensearch and elastic are still compatible enough for this to work) <https://github.com/aws/aws-for-fluent-bit/blob/0046e4553b51dff289334faa1763a9a37c15c193/troubleshooting/debugging.md?plain=1#L639>

Unfortunately I can't find the original documentation page that even hinted this fluentbit version can write to elastic

**Comment by robGetsTheJobDone** [comment]
Amazon is clearly dropping the ball on this. After years of feedback, it's obvious they’re intentionally restricting Fluent Bit outputs on EKS Fargate to force customers into expensive managed services like CloudWatch and Kinesis. Blocking widely-used outputs like http, forward, gelf, and opentelemetry makes no technical sense—except as vendor lock-in. AWS, this isn’t about security or stability, it’s about margin. Let people run real-world observability stacks—Loki, Datadog, SumoLogic, OTEL—without duct-taping sidecars and workarounds. EKS on Fargate is half-baked until this gets fixed.

**Comment by Sharath-Selvaraj** [comment]
Adding support for forward/http outputs in Fargate will really make life much easier for many of us. Right now, we are ending up running extra sidecars or putting extra infra just to push logs to Fluentd or Logstash. This is adding unwanted headache, extra cost, and also going against the whole serverless concept of Fargate. If AWS can kindly enable native support for these plugins, teams like ours can directly plug into existing logging pipelines in a simple straight-forward way, without all these extra adjustments and workarounds. It will be a big help for many customers in real production setups.
